### PR TITLE
fix: #137 Report unhandled rejection promises automatically

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:
- **Purpose**: When a user forgets to `.catch()` a Promise, and this fails, Raygun4Node will capture the error and send it automatically
- **Approach**: Similar to the uncaught exception handler, but for Promises, using `unhandledRejection`.

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)

**Updates**

- Added `process.on` handler.
- Added tags to the `sendSync` method, to help differentiate between "uncaught exceptions" and "unhandled rejections".
- Implemented test.

### Test plan :test_tube:

- Verified using unit tests.
- Verified running simple app:

```js
// Setup Raygun
var raygun = require("raygun");
var raygunClient = new raygun.Client().init({ apiKey: config.Raygun.Key, reportUncaughtExceptions: true });

new Promise(() => {
  throw new Error("test");
}).then(() => {
  // Should not be reached
});
```

image

### Author to check :eyeglasses:
- [x] Project and all contained modules builds successfully
- [x] Self-/dev-tested 
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

**Comment by miquelbeltran** [comment]
rebased to develop

**Review by TheRealAgentK — APPROVED** [review]
Reviewed and I understand now what it's trying to solve and why it's necessary. LGTM.

File: lib/raygun.ts [hunk 1]
```diff
@@ -337,19 +337,32 @@ class Raygun {
...
+      } else {
+        // `reason` type is unknown, wrap in String
+        this.sendSync(`Unhandled Rejection: ${reason}`, ["unhandledRejection"]);
+      }
     });
   }
 
   /**
    * Send error using synchronous transport.
    * Only used to report uncaught exceptions.
```
Comment (nikz) [inline lib/raygun.ts:359]: Tiny: Should we add unhandled promises here? Otherwise it sounds like we're specifically excluding them.

Comment (miquelbeltran) [reply lib/raygun.ts:359]: The function is internal only but good to keep documentation consistent

Comment (miquelbeltran) [reply lib/raygun.ts]: Updated jsdoc comment

**Pull request merged**